codec-http2/src/main/java/io/netty/handler/codec/http2/CompressorHttp2ConnectionEncoder.java

```diff
@@ -73,7 +74,9 @@ public class CompressorHttp2ConnectionEncoder extends DecoratingHttp2ConnectionE
      * with default implementation of {@link StandardCompressionOptions}
      */
     public CompressorHttp2ConnectionEncoder(Http2ConnectionEncoder delegate) {
-        this(delegate, StandardCompressionOptions.brotli(), StandardCompressionOptions.gzip(),
+        this(delegate,
+                Brotli.isAvailable() ? StandardCompressionOptions.brotli() : null,
+                StandardCompressionOptions.gzip(),
                 StandardCompressionOptions.deflate());
     }
```

Comment:
Imho it would be cleaner to do something like this:

```suggestion
                defaultCompressionOptions());
    }
    
    private static CompressionOptions[] defaultCompressionOptions() {
        if (Brotli.isAvailable()) {
            return new CompressionOptions[] {  StandardCompressionOptions.brotli(), StandardCompressionOptions.gzip(), StandardCompressionOptions.deflate() };
        }
        return new CompressionOptions[] { StandardCompressionOptions.gzip(), StandardCompressionOptions.deflate() };
    }
```